### DISCUSSION TITLE
2024-07-31 수요일

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#154] [기록보드]
2024-07-31 수요일 기록보드입니다.

## 여행가고십조(A10) 일일 공통 일과표

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 아침 회의 | |
| 10:00 ~ 11:00 | 기술면접 준비 | |
| 11:00 ~ 12:00 | 집중 개발 time | |
| 12:00 ~ 13:00 | 튜터님 피드백 | <img src="https://github.com/user-attachments/assets/a44bea10-b2f2-4ea7-8abc-888dcd050092" height="300" width="300"/> |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 18:00 | 집중 개발 time | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 저녁 회의 | |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

```
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 내용

### 트러블 슈팅

- 내용

### 내일 계획

- 내용
```

```
// 이미지 테이블
<table>
  <tr>
    <th>이미지 1번 제목</th>
    <th>이미지 2번 제목</th>
  </tr>
  <tr>
    <td><img src="이미지 주소" width="300" height="300"/></td>
    <td><img src="이미지 주소" width="300" height="300"/></td>
  </tr>
</table>
```

**Comment by eunohhh** [comment]
## 개인 기록보드 서식 (오은)

### 오늘 한 일

- onboarding 페이지 기능 및 UI 수정
- 375px 레이아웃 해제
- 스토리 페이지 작성
- 네이버로그인 스크립트 버튼컴포넌트레벨로 내림

### 트러블 슈팅

#### 온보딩 관련
- 튜터님 피드백에 따라 온보딩 fetch 마지막에 한번에 하는 것으로 변경

### 내일 계획

- 스토리 관련 진행

**Comment by Captain-Kim** [comment]
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- Follow 기능 개발 예정
  - 의사코드 : url parameter받은 id 값이 현재 세션의 buddy_id와 일치하면 '프로필 수정' 아니면 '팔로우 하기'
  - '팔로우 하기' 클릭 시 세션이 읽어와 지지 않으면 경고창 -> isOnboarding true까지 확인
- 메인 플로팅 버튼(글쓰기) 예정 

### 트러블 슈팅

- 내용

### 내일 계획

- 내용

**Comment by slsl2** [comment]
## 개인 기록보드 (김민지)
### 오늘 한 일

- 채팅 실시간 프로필 사진 안 뜨는 이슈 해결
- 채팅 목록 페이지 데이터 연결

### 코딩 전에 로직 정리 (채팅 목록 페이지)

**로직**

- buddy_id가 contract.contract_buddy_id와 일치하는
- 모든 contract_id를 찾고, isPending === false && isValidate === true 인 contract_id와 그 contract_trip_id를 필터링
- filtered contract_id마다 있는 모든 contract_buddy_id를 찾고
- 각 buddy_id마다 buddies.buddy_profile_pic를 객체 속 배열로 반환.
- 이런 느낌..
data = [{ contract_id: "", contract_trip_id: "", contract_buddies_profile_pics: ["", "", ""] }, ]
contract_trip_id와 foreign key로 연결된 trips.trip_id를 찾아
- trips.trip_title 반환
- trips.trip_id와 foreign key로 연결된 가장 최근 생성된 messages.message_trip_id 찾기
- messages.message_content, messages.message_created_at 반환

**사용 column**

- contract.contract_trip_id / contract.contract_buddy_id / contract.contract_isPending / contact.contract_isValidate
- trips.trip_id / trips.trip_title
- buddies.buddy_id / buddies.buddy_profile_pic
- messages.message_content / messages.message_created_at / messages.message_trip_id


### 내일 할 일
- 채팅 목록 페이지 데이터 연결 완성
- 채팅 페이지 헤더 데이터 연결
- home 페이지 배너 작업
